bot/pagination.py

```diff
@@ -228,7 +228,10 @@ async def paginate(
         current_page = 0
 
         if not restrict_to_user:
-            restrict_to_user = ctx.author
+            if isinstance(ctx, discord.Interaction):
+                restrict_to_user = ctx.user
+            else:
```

Comment:
I think this can boil down to 
```suggestion
if not restrict_to_user:
    restrict_to_user = getattr(ctx, "user", ctx.author)
```

Reply:
I don't think that would work as `Interaction` doesn't have an author, so it would raise an error (as all the function arguments are evaluated before the function is called)

Reply:
Ho, you're actually right.
The way it works didn't actually match my thoughts, sorry for that

Reply:
> I don't think that would work as `Interaction` doesn't have an author, so it would raise an error (as all the function arguments are evaluated before the function is called)

```suggestion
if not restrict_to_user:
    restrict_to_user = getattr(ctx, "user", None) or ctx.author
```

not the best but it would work with shtlrs idea.